### DISCUSSION TITLE
problem setting up an objective function

### DISCUSSION THREAD

**Discussion by shaicoh3n** [opencobra/cobrapy#1373] [General]
I want to have the objective function of my model (iJO1366 - contains ~2600 reactions) to be minimize the sum of all the fluxes and the sum of 37 slack variables (or artificial fluxes) that are weighted.
<img width="315" alt="obj_func" src="https://github.com/opencobra/cobrapy/assets/89584516/665bf0f1-6657-4078-9e36-da77a454cede">
so the v's are all the fluxes, beta is a scalar and the alpha's are the slack variables/artificial fluxes.
When I try to set this as the objective function it takes a very long time to set up (I gave up after 15 minutes). Is there no faster way to do this than what i'm doing now which is:
model.objective = model.problem.Objective(
        sum(r.flux_expression for r in real_rxns) + beta * sum(r.flux_expression for r in alphas_rxns),
        direction='min')

Thanks to anyone who helps!

**Comment by Midnighter** [comment]
Could you try importing sum from optlang.symbolics, that might be faster. It's quite likely optlang being slow, as it tries to simplify the expression with each added term.

**Reply by shaicoh3n** [reply]
thx, eventually i just let it run without doing anything and it took a little over an hour for ~2500 reactions

**Reply by cdiener** [reply]
The fastest way would be using `Objective.set_linear_coefficients` from optlang but it's odd it takes that long here. We use the symbolic API for models with 250K+ reactions and it usually only takes a few seconds.
